### PR TITLE
✨ [MOOV-2416]: Added Edit Payout functionality

### PR DESCRIPTION
**Pull request by agranillonf**
Added functionality to the "Edit payout" button on the Payout Details modal.

### CODE REVIEW

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/128.

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Looks good!

2 comments:
- I think the beneficiary should still be the predetermined one, and only allow the manually editing if the user requests it.
- We need to add a success toast message that confirms the changes weere made. "Changes applied"

**Review by Publicker — APPROVED** [review]

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/128.

**Comment by agranillonf** [comment]
@pabloi-nf Fixes have been applied.

**Review by arifmatin — APPROVED** [review]

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Thanks @agranillonf 

I can see it doesn't says anymore "Add manually", but it's not showing the beneficiary in the list either.
Is not possible to show it like this?:
image

Another thing is that when there are no changes made and the save button is pressed, the toast still says that changes were made.
We should use a disabled button if there are no changes.
Text color: #73808C
Background color: #E3E5E8

image

**Review by donalnofrixion — CHANGES_REQUESTED** [review]
If I create a payout with a future scheduled date the edit dialog doesn't reflect this.

Screenshot 2023-11-07 at 09 32 14

**Comment by donalnofrixion** [comment]
When on the Edit panel it feels weird that the escape takes you back to the Details panel. Personally I think it should replace the details modal.

**Comment by donalnofrixion** [comment]
Should the 'No changes made button' be visible when first entering the edit modal? I only see the 'Save changes' even if no changes were made.

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
If I press enter when the edit panel shows up it's the same as pressing escape. 
I think the problem is that the back button is the first element to have focus. We should have the amount input as the first element in the tab index order.
image

**Comment by donalnofrixion** [comment]
> > When on the Edit panel it feels weird that the escape takes you back to the Details panel. Personally I think it should replace the details modal.
> 
> @donalnofrixion It makes sense what you say if we assume the user just wants to cancel the whole flow. In that case it's good to speed up the process. However, we don't know really the user intention here. Maybe they pressed the button by mistake and wanted to go back to the details, or they realized the change wasn't necessary and just wanted to proceed to authorise it. In most cases the user expectation is to go back to the previous state, which in this case is with the payout details panel open.

@pabloi-nf That's fair enough. Makes sense

File: packages/components/src/components/ui/organisms/SavePayoutModal/SavePayoutModal.tsx [hunk 1]
```diff
@@ -725,28 +858,33 @@ const CreatePayoutModal: React.FC<CreatePayoutModalProps> = ({
...
+                      disabled={isCreateAndApproveButtonDisabled || !changesMade}
                     >
-                      {isCreateAndApproveButtonDisabled ? (
+                      {!changesMade ? (
+                        <span>No changes made</span>
+                      ) : isCreateAndApproveButtonDisabled ? (
                         <Loader className="h-6 w-6 mx-auto" />
                       ) : (
-                        <span>Create and authorise</span>
+                        <span>{selectedPayout ? 'Save changes' : 'Create and authorise'}</span>
```
Comment (donalnofrixion) [inline packages/components/src/components/ui/organisms/SavePayoutModal/SavePayoutModal.tsx:868]: Should this be on the save button below instead? If the user isn't an Approver this button won't be available.

**Review by donalnofrixion — CHANGES_REQUESTED** [review]
If I login with a user that has the role User I can't edit a payout. The save changes button isn't available

File: packages/components/src/components/functional/SavePayoutModal/SavePayoutModal.tsx [hunk 2]
```diff
@@ -99,15 +111,62 @@ const CreatePayoutModal = ({
...
+  }
+
   return (
     <>
-      <UICreatePayoutModal
+      <UISavePayoutModal
         onDismiss={onDismiss}
         isOpen={isOpen}
-        onCreatePayout={onCreatePayout}
+        onSavePayout={selectedPayout ? onUpdatePayout : onCreatePayout}
```
Comment (donalnofrixion) [inline packages/components/src/components/functional/SavePayoutModal/SavePayoutModal.tsx:165]: I think this should be split into 2 events. onCreatePayout and onSavePayout. The logic isn't very explicit here and might lead to a misunderstanding

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
If I try to remove a description it allows me to save changes, but the changes are not saved.
See video

https://github.com/nofrixion/nofrixion.business/assets/127946572/ee689241-0a12-44d2-b846-6eeb8a3a0df7

**Comment by donalnofrixion** [comment]
@pabloi-nf Just a thought. When editing a payout I think it would be good to be able to approve it at the same time. Similar to creating a payout?

**Comment by pabloi-nf** [comment]
@agranillonf something weird happens here. If I make a change on the edit screen, but then move to another tab or window, the changes dissappear and the data goes back to its original state.

https://github.com/nofrixion/nofrixion.business/assets/127946572/f3955914-d2be-46cc-9c40-bba4ee459a8b

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
The style of the disabled button has to be:
text- color: #73808C
bg-color: #E3E5E8

**Comment by pabloi-nf** [comment]
> @pabloi-nf Just a thought. When editing a payout I think it would be good to be able to approve it at the same time. Similar to creating a payout?

that makes sense @donalnofrixion. My only thought is if a user would expect to be able to authorise it right here, but if it clearly label the buttons that's probably nor a big concern.

We can change the button text from "create" to "save":

image

**Comment by donalnofrixion** [comment]
> > @pabloi-nf Just a thought. When editing a payout I think it would be good to be able to approve it at the same time. Similar to creating a payout?
> 
> that makes sense @donalnofrixion. My only thought is if a user would expect to be able to authorise it right here, but if we clearly label the buttons that's probably not a big concern.
> 
> We can change the button text from "create" to "save":
> 
> image

When I edited a payout it was the first thing I expected to be able to do.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/128.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/128.

**Review by donalnofrixion — APPROVED** [review]

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
The problem where fields are not deleted still happens.

https://github.com/nofrixion/nofrixion.business/assets/127946572/05f17df2-3c3f-449f-b22b-a8e1c0557d90

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/128.

**Review by donalnofrixion — APPROVED** [review]

**Review by pabloi-nf — APPROVED** [review]

**Comment by Publicker** [comment]
crab rave

**Pull request merged**